src/sfCommand.ts

```diff
@@ -26,38 +30,70 @@ export abstract class SfCommand<T> extends Command {
   public static configurationVariablesSection?: HelpSection;
   public static envVariablesSection?: HelpSection;
   public static errorCodes?: HelpSection;
+  public static exposesSensitiveInfo?: boolean;
+
+  private warnings: SfCommand.Warning[] = [];
+
+  /**
+   * Log warning to users. If --json is enabled, then the warning
+   * will be added to the json output under the warnings property.
+   */
+  public warn(input: SfCommand.Warning): SfCommand.Warning {
+    const warning = super.warn(input) as SfCommand.Warning;
+    this.warnings.push(warning);
+    return input;
+  }
 
+  /**
+   * Warn user about sensitive information (access tokens, etc...) before
+   * logging to the console.
+   */
+  public logSensitive(msg: string): void {
+    this.warn(messages.getMessage('warning.security'));
```

Comment:
@mdonnalley shouldn't these be guarded by `exposesSensitiveInfo`? If not, I don't see a need for the property.